src/main/scala/beam/router/RoutingWorker.scala

```diff
@@ -174,19 +175,22 @@ class RoutingWorker(workerParams: R5Parameters) extends Actor with ActorLogging
               )
             } else None
 
-            //run r5 for everything except cars
-            val r5Response = if (request.streetVehicles.exists(_.mode != CAR)) {
-              Some(r5.calcRoute(request.copy(streetVehicles = request.streetVehicles.filter(_.mode != CAR))))
-            } else None
+            val response = if (ghResponse.isEmpty || ghResponse.exists(_.itineraries.isEmpty)) {
+              recallR5ForEmptyGHResponse.incrementAndGet()
+              r5.calcRoute(request)
+            } else if (request.streetVehicles.exists(_.mode != CAR)) {
+              val r5Response =
+                Some(r5.calcRoute(request.copy(streetVehicles = request.streetVehicles.filter(_.mode != CAR))))
+              //combine into one response
+              val resp = Seq(ghResponse, r5Response).collectFirst { case Some(res) => res }.get
```

Comment:
If i am not mistaken, now you can always grap GH response here instead this **collectFirst** construction